contracts/MarginCalculator.sol

```diff
@@ -2,11 +2,280 @@
+     * @return isExcess true if there's excess margin in the vault. In this case, collateral can be taken out from the vault. False if there is insufficient margin and additional collateral needs to be added to the vault to create the position.
+     */
+    function getExcessMargin(Vault memory _vault, address _demonimated)
+        public
+        view
+        returns (uint256 netValue, bool isExcess)
+    {
+        _checkAssetCount(_vault);
+        _checkLongAsset(_vault);
+
+        int256 collateralAmount = _vault.collateralAmounts.length > 0
+            ? FixedPointInt256.uintToInt(_vault.collateralAmounts[0])
+            : 0;
+
+        // Vault contains no short tokens: return collateral value.
+        if (_vault.shortOtokens.length == 0) return (collateralAmount.intToUint(), true);
+
+        // For the currenct version, ensure denominated == short.collateral
+        address shortCollateral = OtokenInterface(_vault.shortOtokens[0]).collateralAsset();
+        require(shortCollateral == _demonimated, "MarginCalculator: Denomintated token should be short.collateral");
+
+        int256 netOtoken = _calculateOtokenNetValue(_vault);
+        int256 totalValue = collateralAmount.add(netOtoken);
+        isExcess = totalValue >= 0;
+        netValue = totalValue.intToUint();
+    }
+
+    /**
+     * @dev Calculate the net value of long token + short token in a vault.
+     * @return netOtoken net worth of long otoken and short otoken
+     */
+    function _calculateOtokenNetValue(Vault memory _vault) internal view returns (int256 netOtoken) {
+        address short = _vault.shortOtokens[0];
+        int256 shortAmount = _vault.shortAmounts.length > 0 ? FixedPointInt256.uintToInt(_vault.shortAmounts[0]) : 0;
+        int256 longAmount = _vault.longAmounts.length > 0 ? FixedPointInt256.uintToInt(_vault.longAmounts[0]) : 0;
+
+        bool expired = now > OtokenInterface(short).expiryTimestamp();
+        bool isPut = OtokenInterface(short).isPut();
+
+        if (!expired) {
+            int256 shortStrike = FixedPointInt256.uintToInt(OtokenInterface(short).strikePrice());
+            int256 longStrike = 0;
+            if (_vault.longOtokens.length > 0)
+                longStrike = FixedPointInt256.uintToInt(OtokenInterface(_vault.longOtokens[0]).strikePrice());
+
+            if (isPut) {
+                netOtoken = _calculateNonExpiredPutNetValue(shortAmount, longAmount, shortStrike, longStrike);
+            } else {
+                netOtoken = _calculateNonExpiredCallNetValue(shortAmount, longAmount, shortStrike, longStrike);
+            }
+        } else {
+            int256 shortCashValue = FixedPointInt256.uintToInt(getExpiredCashValue(short));
+            int256 longCashValue = 0;
+            if (_vault.longOtokens.length > 0)
+                longCashValue = FixedPointInt256.uintToInt(getExpiredCashValue(_vault.longOtokens[0]));
```

Comment:
why not just call it getCashValue if it's also used to get cash value prior to expriy?

Reply:
i.e. cash value is somethign that doesn't change with time, it has no theta

Reply:
This function is only called post expiry. 
But does it make sense to call it `expiredCashValue` ?